AssetRipper.Library/Exporters/Audio/AudioMixerExportCollection.cs

```diff
@@ -0,0 +1,306 @@
+				effect.ObjectHideFlags = HideFlags.HideInHierarchy | HideFlags.HideInInspector;
+				PPtr_AudioMixerEffectController_ effectPPtr = new();
+				effectPPtr.CopyValues(effect.SerializedFile.CreatePPtr(effect));
+				effects[i] = (effect, effectPPtr);
+				AddAsset(effect);
+			}
+			
+			List<Utf8String> pluginEffectNames = ParseNameBuffer(constants.PluginEffectNameBuffer);
+			int pluginEffectIndex = 0;
+			for (int i = 0; i < constants.Effects.Count; i++)
+			{
+				EffectConstant effectConstant = constants.Effects[i];
+				(IAudioMixerEffectController effect, PPtr_AudioMixerEffectController_ effectPPtr) = effects[i];
+
+				IAudioMixerGroupController group = context.groups[effectConstant.GroupConstantIndex];
+				group.Effects_C243.Add(effectPPtr);
+
+				effect.EffectID_C244.CopyValues(constants.EffectGUIDs[i]);
+
+				if (AudioEffectDefinitions.IsPluginEffect(effectConstant.Type))
+				{
+					effect.EffectName_C244.CopyValues(pluginEffectNames[pluginEffectIndex++]);
+				}
+				else
+				{
+					string name = AudioEffectDefinitions.EffectTypeToName(effectConstant.Type) ?? "Unknown";
+					effect.EffectName_C244.String = name;
+					if (name == "Attenuation")
+					{
+						groupsWithAttenuation.Add(group);
+					}
+				}
+
+				bool enableWetMix = (int)effectConstant.WetMixLevelIndex != -1;
+				if (enableWetMix || effect.EffectName_C244 == "Send")
+				{
+					effect.MixLevel_C244.CopyValues(IndexingNewGuid(effectConstant.WetMixLevelIndex, context.indexToGuid));
+				}
+
+				for (int j = 0; j < effectConstant.ParameterIndices.Length; j++)
+				{
+					Parameter param = effect.Parameters_C244.AddNew();
+					// Use a dummy name here. The actual name will be recovered by AssetRipperAudioMixerPostprocessor.
+					param.ParameterName.String = $"Param_{j}";
+					HasAnyEffectParameterNameToRecover = true;
+					param.GUID.CopyValues(IndexingNewGuid(effectConstant.ParameterIndices[j], context.indexToGuid));
+				}
+				
+				if ((int)effectConstant.SendTargetEffectIndex != -1)
+				{
+					effect.SendTarget_C244.CopyValues(effects[effectConstant.SendTargetEffectIndex].Item2);
+				}
+				effect.EnableWetMix_C244 = enableWetMix;
+				effect.Bypass_C244 = effectConstant.Bypass;
+			}
+			
+			// append an Attenuation effect to a group if it has not yet got one,
+			// as Unity doesn't store Attenuation effect if it is the last.
+			foreach (IAudioMixerGroupController group in context.groups)
+			{
+				if (!groupsWithAttenuation.Contains(group))
+				{
+					IAudioMixerEffectController effect = context.virtualFile.CreateAsset<IAudioMixerEffectController>(ClassIDType.AudioMixerEffectController);
+					effect.ObjectHideFlags = HideFlags.HideInHierarchy | HideFlags.HideInInspector;
+					PPtr_AudioMixerEffectController_ effectPPtr = new();
+					effectPPtr.CopyValues(effect.SerializedFile.CreatePPtr(effect));
+					group.Effects_C243.Add(effectPPtr);
```

Comment:
Have you tried exporting multiple times in the same session? The `VirtualSerializedFile` gets deleted after export, possibly resulting in a bad PPtr here.

Reply:
In the new push, Effects_C243 gets cleared at the beginning of every export, so this won't be a problem.